awspub/image.py

```diff
@@ -495,16 +495,22 @@ def public(self) -> None:
 
         # handle marketplace publication
         if self.conf["marketplace"]:
-            logger.info(f"marketplace version request for {self.image_name}")
-            # image needs to be in us-east-1
-            ec2client: EC2Client = boto3.client("ec2", region_name="us-east-1")
-            image_info: Optional[_ImageInfo] = self._get(ec2client)
-            if image_info:
-                im = ImageMarketplace(self._ctx, self.image_name)
-                im.request_new_version(image_info.image_id)
+            # the "marketplace" configuration is only valid in the "aws" partition
+            if boto3.client("ec2").meta.partition == "aws":
```

Comment:
do we not have a shared client somewhere? i'm not sure exactly how client works but i suspect it initiates a connection each time. we should avoid repeatedly creating it. this might be better suited for a follow up PR though

Reply:
we don't (yet). we have to handle one client for each region. currently that client gets created on the fly. definitely room for improvements there.